rialto_airflow/mais.py

```diff
@@ -58,21 +62,20 @@ def get_token(client_id: str, client_secret: str, base_url: str) -> Optional[str
             logger.info("Successfully obtained access token.")  # Log success
             return access_token
         else:
-            logger.error(
-                f"No 'access_token' found in JSON response: {data}"
-            )  # Log specific error
-            return None
+            err_msg = f"No 'access_token' found in JSON response: {data}"
+            logger.error(err_msg)  # Log specific error
+            raise TokenFetchError(err_msg)
```

Comment:
this sprang from the type checker complaining that a subsequent call[^1] treated the return value from this function as if it would be non-`None`.  could've made the value optional and raised in the receiving function, but a cursory read through of the code made me think that a bunch of downstream stuff assumed success here, so i figured maybe better to just fail early?  happy to revisit if `return None` was better for reasons i didn't understand.

[^1]:  `current_orcid_users` in `orcid_integration_stats` task in `publish_orcid` DAG/module